integration/test/metrics_number_dimension/metrics_number_dimension_test.go

```diff
@@ -0,0 +1,133 @@
+
+//go:build linux && integration
+// +build linux,integration
+
+package metrics_number_dimension
+
+import (
+	"context"
+	"fmt"
+	"github.com/aws/amazon-cloudwatch-agent/integration/test"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch/types"
+	"log"
+	"strings"
+	"testing"
+	"time"
+)
+
+const configOutputPath = "/opt/aws/amazon-cloudwatch-agent/bin/config.json"
+const configJSON = "/config.json"
+const namespace = "MetricNumberDimensionTest"
+const instanceId = "InstanceId"
+const append = "append"
+
+//Let the agent run for 2 minutes. This will give agent enough time to call server
+const agentRuntime = 2 * time.Minute
+
+var targetString = "max MaxDimensions %v is less than than number of dimensions %v thus only taking the max number"
+
+type input struct {
+	dataInput            string
+	findTarget           bool
+	inputNumberDimension int
+	metricName           string
+}
+
+type metric struct {
+	name  string
+	value string
+}
+
+//Must run this test with parallel 1 since this will fail if more than one test is running at the same time
+//This test uses a pem file created for the local stack endpoint to be able to connect via ssl
+func TestBundle(t *testing.T) {
+
+	parameters := []input{
+		{dataInput: "resources/10_dimension",
+			findTarget:           false,
+			inputNumberDimension: 10,
+			metricName:           "mem_used_percent"},
+		// @TODO add when https://github.com/aws/amazon-cloudwatch-agent/pull/361 is merged
+		// {dataInput: "resources/30_dimension", findTarget: false, inputNumberDimension: 30, metricName: "mem_used_percent"},
+		// {dataInput: "resources/35_dimension", findTarget: true, inputNumberDimension: 35, metricName: "mem_used_percent"},
+	}
+
+	for _, parameter := range parameters {
+		//before test run
+		log.Printf("resource file location %s find target %t input number dimension %d metric name %s",
+			parameter.dataInput, parameter.findTarget, parameter.inputNumberDimension, parameter.metricName)
+
+		targetString = fmt.Sprintf(targetString, 30, parameter.inputNumberDimension)
+
+		t.Run(fmt.Sprintf("resource file location %s find target %t", parameter.dataInput, parameter.findTarget), func(t *testing.T) {
+			test.CopyFile(parameter.dataInput+configJSON, configOutputPath)
+			test.StartAgent(configOutputPath)
+			time.Sleep(agentRuntime)
+			log.Printf("Agent has been running for : %s", agentRuntime.String())
+			test.StopAgent()
+
+			// test for target string
+			output := test.ReadAgentOutput(agentRuntime)
+			containsTarget := outputLogContainsTarget(output)
+			if (parameter.findTarget && !containsTarget) || (!parameter.findTarget && containsTarget) {
+				t.Errorf("Find target is %t contains target is %t", parameter.findTarget, containsTarget)
+			}
+
+			// test for cloud watch metrics
+			cxt := context.Background()
+			dimensionFilter := buildDimensionFilterList(parameter.inputNumberDimension - 1)
```

Comment:
Can you explain why we pass in `- 1` here? I see that we add 1 back to it inside the `buildDimensionFilterList`, so this feels confusing to me.